connectors/src/connectors/confluence/index.ts

```diff
@@ -49,479 +49,516 @@ import { getAccessTokenFromNango } from "@connectors/lib/nango_helpers";
-      isExpandable: true,
-    });
-  });
+    const contentNodes = spaceContentNodes.concat(pageContentNodes);
+    return new Ok(contentNodes);
+  }
 
-  const pageContentNodes: ContentNode[] = await concurrentExecutor(
-    confluencePages,
-    async (page) => {
-      let parentId: string;
-      let parentType: "page" | "space";
+  async retrieveContentNodeParents({
+    internalId,
+  }: {
+    internalId: string;
+    memoizationKey?: string;
+  }): Promise<Result<string[], Error>> {
+    const confluenceId = getIdFromConfluenceInternalId(internalId);
 
-      if (page.parentId) {
-        parentId = page.parentId;
-        parentType = "page";
-      } else {
-        parentId = page.spaceId;
-        parentType = "space";
+    if (isConfluenceInternalPageId(internalId)) {
+      const currentPage = await ConfluencePage.findOne({
+        attributes: ["pageId", "parentId", "spaceId"],
+        where: {
+          connectorId: this.connectorId,
+          pageId: confluenceId,
+        },
+      });
+
+      if (!currentPage) {
+        return new Err(new Error("Confluence page not found."));
+      }
+
+      // If the page does not have a parentId, return only the spaceId.
+      if (!currentPage.parentId) {
+        return new Ok([makeConfluenceInternalSpaceId(currentPage.spaceId)]);
       }
-      const isExpandable = await checkPageHasChildren(connectorId, page.pageId);
 
-      return createContentNodeFromPage(
-        { id: parentId, type: parentType },
-        connectorState.url,
-        page,
-        isExpandable
+      const parentIds = await getConfluencePageParentIds(
+        this.connectorId,
+        currentPage
       );
-    },
-    { concurrency: 8 }
-  );
+      return new Ok(parentIds);
+    }
 
-  const contentNodes = spaceContentNodes.concat(pageContentNodes);
-  return new Ok(contentNodes);
-}
+    return new Ok([]);
+  }
 
-export async function retrieveConfluenceContentNodeParents(
-  connectorId: ModelId,
-  internalId: string
-): Promise<Result<string[], Error>> {
-  const confluenceId = getIdFromConfluenceInternalId(internalId);
+  async pause(): Promise<Result<undefined, Error>> {
+    const connector = await ConnectorResource.fetchById(this.connectorId);
+    if (!connector) {
+      logger.error({ connectorId: this.connectorId }, "Connector not found.");
+      return new Err(new Error("Connector not found"));
+    }
 
-  if (isConfluenceInternalPageId(internalId)) {
-    const currentPage = await ConfluencePage.findOne({
-      attributes: ["pageId", "parentId", "spaceId"],
-      where: {
-        connectorId,
-        pageId: confluenceId,
-      },
-    });
+    await connector.markAsPaused();
+    const r = await stopConfluenceSyncWorkflow(this.connectorId);
+    if (r.isErr()) {
+      return r;
+    }
 
-    if (!currentPage) {
-      return new Err(new Error("Confluence page not found."));
+    return new Ok(undefined);
+  }
+
+  async unpause(): Promise<Result<undefined, Error>> {
```

Comment:
What's the difference between `unpause` and `resume`?

Reply:
Resume is what you do to undo "stop". Unpause is how you undo "pause". This needs to be cleaned up (ideally stop/resume should go away or replace pause/unpause)